client/components/main/index.jsx

```diff
@@ -3,17 +3,36 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classnames = require( 'classnames' );
+import React from 'react';
+import { connect }	from 'react-redux';
+import classnames from 'classnames';
 
-module.exports = React.createClass( {
+/**
+ * Internal dependencies
+ */
+import { shouldViewBeVisible } from 'state/first-view/selectors';
+
+const Main = React.createClass( {
 	displayName: 'Main',
 
 	render: function() {
+		const classes = classnames( this.props.className, 'main', {
+			'is-first-view-visible': this.props.isFirstViewVisible
+		} );
+
 		return (
-			<main className={ classnames( this.props.className, 'main' ) } role="main">
+			<main className={ classes } role="main">
 				{ this.props.children }
 			</main>
 		);
 	}
 } );
+
+export default connect(
```

Comment:
@ehg This approach works well. However, it requires me to modify every place that uses `<Main>` to wrap it in a `<Provider>` -- is that just the way it has to be, or do you have any suggestions for avoiding this?

For example, `/help` fails with the following:

```
Uncaught Invariant Violation: Could not find "store" in either the context or props of "Connect(Main)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(Main)".
```